### DISCUSSION TITLE
Tools and demo for transients

### DISCUSSION THREAD

**Discussion by samuel-clarke** [magenta/ddsp#112] [Ideas]
I've been working with this library for a few weeks now, and I thought there was nothing I couldn't do with this. But I've recently been trying to synthesize sounds where the transient component is essential. The core.oscillator_bank, FilteredNoise, and the current SpectralLoss don't seem to be sufficient.

Perhaps the tools I need are already there, and I just don't know what I'm doing. Either way, it would be great if you included an instrument where the transients are really important, such as a drum set, somewhere in your demos or tutorials. Thanks for all your great work!

**Comment by jesseengel** [comment]
Agreed! Transients are definitely an important avenue for more research and there are a few ways that might work to go about it (For example, adding the oscillator output to a smaller more conventional CNN that generates waveform directly), but we haven't had time to focus on this yet.

**Comment by erl-j** [comment]
It might be valuable for you to check out the following paper.

https://go.gale.com/ps/anonymous?id=GALE%7CA168285300&sid=googleScholar&v=2.1&it=r&linkaccess=abs&issn=01489267&p=AONE&sw=w&u=googlescholar&asid=e2e13b99070d7294800f3b3a7a31e184&mg=true

It details a solution for the transient problem.
